src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/FixAllContextHelper.cs

```diff
@@ -137,8 +183,90 @@ public static string GetDefaultFixAllTitle(FixAllContext fixAllContext)
                 FixAllScope.Document => string.Format(WorkspaceExtensionsResources.Fix_all_0_in_1, diagnosticId, triggerDocument!.Name),
                 FixAllScope.Project => string.Format(WorkspaceExtensionsResources.Fix_all_0_in_1, diagnosticId, triggerProject.Name),
                 FixAllScope.Solution => string.Format(WorkspaceExtensionsResources.Fix_all_0_in_Solution, diagnosticId),
+                FixAllScope.ContainingMember => string.Format(WorkspaceExtensionsResources.Fix_all_0_in_Containing_Member, diagnosticId),
+                FixAllScope.ContainingType => string.Format(WorkspaceExtensionsResources.Fix_all_0_in_Containing_Type, diagnosticId),
                 _ => throw ExceptionUtilities.UnexpectedValue(fixAllScope),
             };
         }
+
+        public static async Task<ImmutableDictionary<Document, ImmutableArray<TextSpan>>> GetDocumentsAndSpansForContainingMemberOrTypeAsync(
+            Document document, FixAllScope fixAllScope, TextSpan triggerSpan, CancellationToken cancellationToken)
+        {
+            Contract.ThrowIfFalse(fixAllScope is FixAllScope.ContainingMember or FixAllScope.ContainingType);
+
+            var decl = await GetContainingMemberOrTypeDeclarationAsync(document, fixAllScope, triggerSpan, cancellationToken).ConfigureAwait(false);
+            if (decl == null)
```

Comment:
when does this happen?  do we offer such a fixallscope if the user is not in sucha  scope?

Reply:
This will happen for file level diagnostics, say unnecessary usings. We will not show such a fix scope to users for a code fix to such diagnostics. That filtering is done here: https://github.com/dotnet/roslyn/pull/59989#discussion_r820627128